### PR TITLE
Added support for consumer and producer to have key and value

### PR DESCRIPTION
**Pull request by saxenakshitiz**

### CODE REVIEW

**Review by laxmanchekka — DISMISSED** [review]

**Review by laxmanchekka — APPROVED** [review]

File: event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java [hunk 1]
```diff
@@ -1,41 +1,32 @@
...
 
 import java.time.Duration;
+import java.util.List;
 
 /**
  * While this is similar to Kafka consumer. The key difference is that the consumer is initialized
  * per topic unlike Kafka where one can consume events across all topics.
  */
-public interface EventConsumer<T> {
+public interface EventConsumer<K, V> {
```
Comment (tim-mwangi) [inline event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java:10]: Do we care about backwards compatibility where the old interface would be used by some of the dependents and they would spit out data to be consumed by ones using the new API?

Comment (saxenakshitiz) [reply event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java:10]: From kafka perspective, key and value type are always needed. In the previous version, `<K>` was hard-coded to `byte[]` while `<V>` was being provided.
If someone moves to new version of event store, then they will have explicitly pass `byte[]` for `<K>`. Apart from that there will be no compatibility either on consumer or producer side.

Comment (tim-mwangi) [reply event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java:10]: >Apart from that there will be no compatibility either on consumer or producer side.

You mean there's no issue on a consumer using the old version and a producer using the new one?

Comment (saxenakshitiz) [reply event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java:10]: Yes, they will work without any issue.

Comment (tim-mwangi) [reply event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java:10]: ok.

**Review by tim-mwangi — APPROVED** [review]

**Review by surajpuvvada — APPROVED** [review]

File: event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java (hunk 1 above)
Comment (aaron-steinfeld) [reply event-store/src/main/java/org/hypertrace/core/eventstore/EventConsumer.java:10]: :( we always care about backwards compatibility about APIs! not just between producers and consumers, but between multiple modules in the same JVMs.

**Pull request merged**